### PR TITLE
club find use search endpoint

### PR DESCRIPTION
**Pull request by j-martin**
At this point this is a proposal.

The rationale to keep the in-client filtering is that the [search DSL](https://help.clubhouse.io/hc/en-us/articles/360000046646-Search-Operators) is still quite limited and does not support sorting for example.

I stripped out the `listStories` call, but we may want to keep it as the default if no argument is passed.

### CODE REVIEW

**Comment by andjosh** [comment]
Thanks! I'll take a look through this today.

**Comment by andjosh** [comment]
I think this makes sense (to use the official search endpoint). I agree that the search DSL is a bit lacking (which is one reason I built this CLI - to use exact filtering).

Is the search endpoint it faster than calling all and filtering? (I hope so! 😄 )

What do you think about keeping the `listStories` as the search method that is used if `club find` is called with flags and otherwise allowing `club find` to be called with a text argument?

i.e.

```sh
club find --label foo --type bug # searches using the listStories method
```

```sh
club find "label:foo" # searches using the searchStories method
```

**Comment by j-martin** [comment]
It's much, much, faster with their API. The time it takes clubhouse-cli to work in our workspace (with 10,000s of stories) is the reason why I added the search endpoint to the clubhouse-lib.

Regarding the flag (list stories) vs arguments (search stories), there are still flags that would make sense to preserve. E.g. sorting and other flags not supported by the DSL.

In this case: `club find --label foo --type bug 'label:foo'` I think should use the searchStory where `club find --label foo --type bug` should use the listStory endpoint. Maybe add a log message to clarify it to the user.

Flags like `--label foo` should arguably be translated into `label:foo`, but we should probably do that at a later time.

**Comment by andjosh** [comment]
Agreed, I like that path.

Glad it's faster and looking forward to it!

**Comment by j-martin** [comment]
@andjosh please have a look. :)

File: src/bin/club-find [hunk 1]
```diff
@@ -39,6 +46,9 @@ module.exports = program = program
 spin.setSpinnerString(27);
 
 const main = async () => {
+    if (program.args.length) {
+        log('Using search story API since search operators were passed ...');
```
Comment (andjosh) [inline src/bin/club-find]: Can we have this log message respect the `--quiet` flag?

File: src/bin/club-find [hunk 2]
```diff
@@ -3,14 +3,21 @@ const spin      = require('../lib/spinner.js')('Finding... %s ');
...
+let program = require('commander');
 
 module.exports = program = program
-    .description('Search through clubouse stories')
+    .description(`Search through clubhouse stories. Arguments (non-flag/options) will
+  be passed to Clubhouse story search API as search operators. Note that passing search
+  operators and options (e.g. --owner yourself) will use the options as extra filtering
+  in the client.
+
+  Refer to https://help.clubhouse.io/hc/en-us/articles/360000046646-Search-Operators
```
Comment (andjosh) [inline src/bin/club-find]: This is very helpful!

File: src/bin/club-find (hunk 1 above)
Comment (j-martin) [reply src/bin/club-find]: @andjosh updated.

**Comment by andjosh** [comment]
@j-martin do we want to persist the search args in the saved workspace version (if saving)?

Ref: https://github.com/andjosh/clubhouse-cli/pull/55/files#diff-457cffedf40d6a636998017b8457297eR65

**Comment by j-martin** [comment]
Yes, that would be the sane thing to do. Do you want to add it?

**Comment by j-martin** [comment]
@j-martin I added it to the workspace options.

**Comment by andjosh** [comment]
Released in `v1.13.0`. Thanks @j-martin!

**Pull request merged**